src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -244,7 +245,7 @@ public override void ExecuteSiteRecoveryCmdlet()
                         licenseType = providerSpecificDetails.LicenseType;
                     }
 
-                    if (string.IsNullOrEmpty(this.RecoveryAvailabilitySet))
+                    if (!this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.RecoveryAvailabilitySet)))
```

Comment:
You can use the IsParameterBound pattern, which is much less awkward and verbose: https://github.com/Azure/azure-powershell/blob/1bcbe7b1f7a3323ac98f7754ba03eeb6b45e79f2/src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorBackendObject.cs#L86